pkg/compactor/job.go

```diff
@@ -174,6 +175,10 @@ func jobWaitPeriodElapsed(ctx context.Context, job *Job, waitPeriod time.Duratio
 			continue
 		}
 
+		if skipFutureMaxTime && timestamp.Time(meta.MaxTime).After(threshold) {
+			return false, meta, nil
+		}
+
```

Comment:
Note for the reviewers: this is the seed of the changes in the PR. The rest are mechanical changes to support the new feature flag.